Automattic/pocket-casts-android
# Adds Fastlane release cycle using Buildkite

**Please note that these PRs will break the current release flow, so we'll keep them as draft until we are ready to merge all of them.**

This PR builds on #241 and introduces the necessary lanes and Buildkite pipeline to do a full release cycle. In detail:

* Adds release cycle lanes `code_freeze`, `complete_code_freeze` & `finalize_release`.
* Adds release build to Buildkite.
* Refactors `build_and_upload_beta` into `build_and_upload_to_play_store` which internally figures out if this is a beta or a production build. This simplifies the rest of our lanes and if I am not mistaken, we had an issue/discussion about this in another repo. I can't remember where though, so I can't link it.
* Updates the `CHANGELOG.md` format to work with our tooling so we can extract the release note during `code_freeze`. I intend to open a separate PR for this to let the developers know about the change and weigh in. It's worth noting that this is the format that's used in [Pocket Casts iOS](https://github.com/Automattic/pocket-casts-ios/blob/trunk/CHANGELOG.md) presumably for the same reason.

Note that there is a PR adding a release scenario for Pocket Casts Android in `mobile-toolbox`. Feel free to use that scenario as a guide.

**To test**

I think it's best to skip the testing step in this PR. I tested the `code_freeze` & `complete_code_freeze` lanes by disabling destructive actions and made sure we were able to create a release draft in Play Store and a GitHub draft using Buildkite. It's not likely there are other issues, but if there are, we can always manually handle them and fix in future PRs. So, in my opinion, we should merge this and then run a full release cycle when we are ready.

# Checklist
